[Doc] Create a contact section and cleanup the introduction (#6613)
PyCQA/pylint: doc/index.rst

Pylint documentation
====================

> Pylint is a `static code analyser`_ for python 2 or 3. Only python interpreter above 3.7.2 are supported
> in the latest version.

Reviewer: ```suggestion
Pylint is a `static code analyser`_ for Python 3. The latest version supports Python 3.7.2 and newer.
```

Author: You can use pylint for python 2 : it's version 1.9.6 
```suggestion
Pylint is a `static code analyser`_ for python 2 or 3. Only python interpreter above 3.7.2 are still maintained currently.
```

Reviewer: Not a super issue for me, but let's not mention Python 2 in the first line of our index. Small contribution to the general `Python` ecosystem to nudge everybody towards 3 😄

Author: I looked at mypy's doc for an example, this is why I changed it : https://mypy.readthedocs.io/en/stable/index.html

I don't think Guido can be suspected of trying to incite people to keep/start using python 2 😄

Reviewer: I think the difference is that `mypy` can actually check `2.7` code although you need to run it using `3.6`. I don't actually know this: but do we still support checking `2.7` code? It feels like that would violate a lot of assumptions in our checkers which wouldn't work.

Author: We have old versions that do check python 2, some versions only python 2. If you pip install pylint you'll get a version compatible with your interpreter. I did not try to check if pylint 2.14 can check python 2 code, probably not as you say.

Reviewer: ```suggestion
Pylint is a `static code analyser`_ for Python 2 or 3. The latest version supports Python 3.7.2 and above.
```

> Pylint analyse your code without actually running it. It checks for errors, enforce a coding
> standard, looks for `code smells`_, and can make suggestions about how the code could be refactored.

Reviewer: ```suggestion
Pylint analyses your code without actually running it. It checks for errors, enforces a coding
standard, looks for `code smells`_, and can make suggestions about how the code could be refactored.
```

Project that you might want to use alongside pylint include flake8_ (faster and simpler checks
with very few false positives), mypy_, pyright_ or pyre_ (typing checks), bandit_ (security
oriented checks), black_ and isort_ (auto-formatting).

.. _flake8: https://gitlab.com/pycqa/flake8/
.. _bandit: https://github.com/PyCQA/bandit
.. _mypy: https://github.com/python/mypy
.. _pyright: https://github.com/microsoft/pyright
.. _pyre: https://github.com/facebook/pyre-check
.. _black: https://github.com/psf/black
.. _isort: https://pycqa.github.io/isort/
.. _`static code analyser`: https://en.wikipedia.org/wiki/Static_code_analysis
.. _`code smells`: https://martinfowler.com/bliki/CodeSmell.html

Pylint can infer actual values from your code using it's internal code representation (astroid).

> If your code is ``import logging as argparse``, Pylint can knows that ``argparse.error(...)``

Reviewer: ```suggestion
If your code is ``import logging as argparse``, Pylint will know that ``argparse.error(...)``
```

is in fact a logging call and not an argparse call.

Pylint isn't smarter than you: it may warn you about things that you have
conscientiously done or checks for some things that you don't care about.
During adoption, especially in a legacy project where pylint was never enforced,
it's best to start with the ``--errors-only`` flag, then disable
convention and refactor message with ``--disable=C,R`` and progressively
re-evaluate and re-enable messages as your priorities evolve.

> .. toctree::

Reviewer: Let's remove this `toctree` as well. It doesn't look very pretty, see:
https://pylint--6613.org.readthedocs.build/en/6613/index.html

Author: I mean I'm ok with that but how do we tell sphinx that it needs to display this toctree and include files ?

Reviewer: Can't you hide a `toctree` with an attribute. `hidden: True` I believe? I think I use it in `pydocstringformatter`.

Author: Thanks for the tip 😄 This is indeed a lot better

Pylint documentation
====================

Pylint is a `static code analyser`_ for Python 2 or 3. The latest version supports Python
3.7.2 and above.

Pylint analyses your code without actually running it. It checks for errors, enforces a coding
standard, looks for `code smells`_, and can make suggestions about how the code could be refactored.

> Project that you might want to use alongside pylint include flake8_ (faster and simpler checks

Reviewer: ```suggestion
Projects that you might want to use alongside pylint include flake8_ (faster and simpler checks
```

with very few false positives), mypy_, pyright_ or pyre_ (typing checks), bandit_ (security
oriented checks), black_ and isort_ (auto-formatting), autoflake_ (automated removal of
unused import or variable), pyupgrade_ (automated upgrade to newer python syntax) and

> pydocstringformatter_ (automated pep257).

Reviewer: 😄

Author: It's your last chance to give this tool a cool name, Daniel.

Reviewer: Ship has sailed I think 😄 And I really don't know anything better.

.. _flake8: https://gitlab.com/pycqa/flake8/
.. _bandit: https://github.com/PyCQA/bandit
.. _mypy: https://github.com/python/mypy
.. _pyright: https://github.com/microsoft/pyright
.. _pyre: https://github.com/facebook/pyre-check
.. _black: https://github.com/psf/black
.. _autoflake: https://github.com/myint/autoflake
.. _pyupgrade: https://github.com/asottile/pyupgrade
.. _pydocstringformatter: https://github.com/DanielNoord/pydocstringformatter
.. _isort: https://pycqa.github.io/isort/
.. _`static code analyser`: https://en.wikipedia.org/wiki/Static_code_analysis
.. _`code smells`: https://martinfowler.com/bliki/CodeSmell.html

> Pylint can infer actual values from your code using it's internal code representation (astroid).

Reviewer: ```suggestion
Pylint can infer actual values from your code using its internal code representation (astroid).
```
